java-checks/src/main/java/org/sonar/java/checks/ConstantMathCheck.java

```diff
@@ -0,0 +1,188 @@
+          addIssue(tree, String.format("Remove this silly call to \"Math.%s\"", methodTree.symbol().name()));
+        }
+      }
+    }
+  }
+
+  private boolean isTruncation(MethodInvocationTree methodTree) {
+    return TRUNCATION_METHODS.anyMatch(methodTree) && isCastFromIntegralToFloating(removeParenthesis(methodTree.arguments().get(0)));
+  }
+
+  private boolean isConstantWithZero(MethodInvocationTree methodTree) {
+    return CONSTANT_WITH_ZERO_METHODS.anyMatch(methodTree) && isFloatingZero(methodTree.arguments().get(0));
+  }
+
+  private boolean isConstantWithZeroOrOne(MethodInvocationTree methodTree) {
+    return CONSTANT_WITH_ZERO_OR_ONE_METHODS.anyMatch(methodTree) && isFloatingZeroOrOne(methodTree.arguments().get(0));
+  }
+
+  private boolean isCastFromIntegralToFloating(ExpressionTree tree) {
+    Type resultType = tree.symbolType();
+    // explicit cast
+    if (tree.is(Tree.Kind.TYPE_CAST) && isIntegral(getInnerType(((TypeCastTree) tree).expression())) && (resultType.is("double") || resultType.is("float"))) {
+      return true;
+    }
+    // implicit cast
+    return isIntegral(resultType);
+  }
+
+  private boolean isIntegral(Type type) {
+    return type.isPrimitive() && !type.is("double") && !type.is("float");
+  }
+
+  private boolean isIntegralOne(ExpressionTree tree) {
+    Long value = LiteralUtils.longLiteralValue(tree);
+    return value != null && value == 1;
+  }
+
+  private Type getInnerType(ExpressionTree tree) {
+    ExpressionTree result = tree;
+    while (true) {
+      if (result.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
+        result = ((ParenthesizedTree) result).expression();
+      } else if (result.is(Tree.Kind.TYPE_CAST)) {
+        result = ((TypeCastTree) result).expression();
+      } else {
+        return result.symbolType();
+      }
+    }
+  }
+
+  private ExpressionTree removeParenthesis(ExpressionTree tree) {
+    ExpressionTree result = tree;
+    while (result.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
+      result = ((ParenthesizedTree) result).expression();
+    }
+    return result;
+  }
+
+  private boolean isFloatingZero(ExpressionTree tree) {
+    Double value = parseDouble(removeParenthesis(tree));
+    return value != null && value == 0.0d;
+  }
+
+  private boolean isFloatingZeroOrOne(ExpressionTree tree) {
+    // current implementation of parseDouble returns null for all value but 0 and 1, so there is no need to check the actual returned value.
+    return parseDouble(removeParenthesis(tree)) != null;
+  }
+
+  @CheckForNull
+  private Double parseDouble(ExpressionTree tree) {
+    if (tree.is(Tree.Kind.DOUBLE_LITERAL, Tree.Kind.FLOAT_LITERAL)) {
+      String value = ((LiteralTree) tree).value();
+      if (value.equals("0.0") || value.equals("0.0d") || value.equals("0.0f")) {
+        return 0.0d;
+      } else if (value.equals("1.0") || value.equals("1.0d") || value.equals("1.0f")) {
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Move the "1.0" string literal on the left side of this string comparison.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Move the "1.0f" string literal on the left side of this string comparison.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Move the "1.0d" string literal on the left side of this string comparison.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)